tests/main/snap-set/task.yaml

```diff
@@ -0,0 +1,76 @@
+summary: Check that `snap set` runs apply-config hook.
+
+systems: [-ubuntu-core-16-64]
+
+prepare: |
+    echo "Build basic test package (without hooks)"
+    snapbuild $TESTSLIB/snaps/basic .
+    snap install --dangerous basic_1.0_all.snap
+
+    echo "Build basic hooks package"
+    snapbuild $TESTSLIB/snaps/basic-hooks .
+    snap install --dangerous basic-hooks_1.0_all.snap
+
+    echo "Build failing hooks package"
+    snapbuild $TESTSLIB/snaps/failing-config-hooks .
+    snap install --dangerous failing-config-hooks_1.0_all.snap
+
+    # We want to use the in-tree snap-exec, not the one in the core snap. To
```

Comment:
I actually had that in one of my PRs so I cherry-picked it: https://github.com/snapcore/snapd/pull/1872